crap-advisor-frontend/src/core/components/RestroomMarker/index.tsx

```diff
@@ -1,58 +1,92 @@
-import React, {useEffect, useMemo, useState} from "react";
-import {Marker, Tooltip, useMap} from "react-leaflet";
-import L from "leaflet";
-import {Restroom} from "../../../app/typings";
-import {IMapProps} from "../Map";
-import {LatLngExpression} from 'leaflet'
-import {toLatLngExpression} from "../../../utils";
-import {defaultRestroomMarkerImage, restroomMarkerImages} from "../../../app/constants";
-import "./index.css";
-import "../../../style/main.css";
-
-interface IRestroomMakerProps extends Pick<IMapProps, 'selectRestroom'> {
-    restroom: Restroom
+import React, { useEffect, useMemo, useState } from 'react';
+import { Marker, Tooltip, useMap } from 'react-leaflet';
+import { useNavigate } from 'react-router-dom';
+
+import L from 'leaflet';
+import { LatLngExpression } from 'leaflet';
+
+import {
+  defaultRestroomHue,
+  defaultRestroomMarkerImage,
+  restroomMarkerImages,
+} from 'app/constants';
+import { Restroom } from 'app/typings';
+import { useAppDispatch, useAppSelector } from 'app/typings/redux';
+
+import { start } from 'store/reviews/slice';
+
+import { toLatLngExpression } from 'utils';
+
+import 'style/main.css';
+
+import './index.css';
+
+interface IRestroomMakerProps {
+  restroom: Restroom;
 }
 
-const RestroomMarker = ({restroom, selectRestroom}: IRestroomMakerProps): JSX.Element => {
-    const [image, setImage] = useState(defaultRestroomMarkerImage)
-    const [rating, setRating] = useState<number | null>(null)
+const RestroomMarker = ({ restroom }: IRestroomMakerProps): JSX.Element => {
+  const [image, setImage] = useState(defaultRestroomMarkerImage);
+  const [rating, setRating] = useState<number | null>(null);
+  const { currentRestroom } = useAppSelector((state) => state.REVIEWS);
+  const navigate = useNavigate();
+  const map = useMap();
+  const dispatch = useAppDispatch();
+  const changeSidebarColors = React.useCallback(() => {
```

Comment:
i'll talk about it later...

Reply:
Don't tread on my sweet React memoization! Better stay out of.

Reply:
man this simple map app looks like a horrible spaghetti already